### PR TITLE
Local docker env elasticsearch cluster

### PR DESCRIPTION
**Pull request by kwhickey**
**Description:**
Run a single-node elasticsearch cluster as part of the local docker environment setup for usaspending.

**Technical details:**
Use the official elasticsearch docker image at version 6.2.4, and mount to a data directory the user needs to create, to preserve ES data.

**Requirements for PR merge:**

1. N/A Unit & integration tests updated
2. N/A API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - N/A Frontend <OPTIONAL>
    - [x] Operations <OPTIONAL>
    - N/A Domain Expert <OPTIONAL>
4. N/A Matview impact assessment completed
5. N/A Frontend impact assessment completed
6. N/A Data validation completed
7. [x] Appropriate Operations ticket(s) created
8. [x] Jira Ticket [OPS-274](https://federal-spending-transparency.atlassian.net/browse/OPS-274):
    - [x] Link to this Pull-Request
    - N/A Performance evaluation of affected (API | Script | Download)
    - N/A Before / After data comparison

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -13,3 +13,10 @@ POSTGRES_CLUSTER_DIR=../docker_db
 
 # MATVIEW_SQL_DIR has to be inside of the project (check the docker-compose file)
 MATVIEW_SQL_DIR=matview_sql
+
+# Where to connect to elasticsearch.
+# Should include scheme (http:// or https://, host, and port (if different than 80 or 443 for HTTP/HTTPS)
+ES_HOSTNAME=http://usaspending-es:9200
+
+# POSTGRES_CLUSTER_DIR should point to a path outside, or it will be mounted w/Docker
```
Comment (tony-sappe) [inline .env]: POSTGRES_CLUSTER_DIR -> ES_CLUSTER_DIR

File: docker-compose.yaml [hunk 2]
```diff
@@ -66,3 +67,24 @@ services:
...
       DATABASE_URL: postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/data_store_api
+      ES_HOSTNAME: ${ES_HOSTNAME}
+
+  usaspending-es:
+    image: docker.elastic.co/elasticsearch/elasticsearch:6.2.4
+    environment:
+    - cluster.name=usaspending
+    - network.host=0.0.0.0
+    - bootstrap.memory_lock=true
+    - "ES_JAVA_OPTS=-Xms1536m -Xmx1536m"
```
Comment (tony-sappe) [inline docker-compose.yaml]: Do we want to make this an env var? 
Should we highlight this in the documentation to ensure people are aware of the memory needs for this container?

File: README.md [hunk 3]
```diff
@@ -63,28 +63,21 @@ See below for basic setup instructions. For help with Docker Compose:
...
-```
+- `docker-compose up usaspending-es` will create and start a single-node Elasticsearch cluster, using the `ES_CLUSTER_DIR` specified in the `.env` configuration file. We recommend using a folder outside of the usaspending-api project directory so it does not get copied to other containers.
+
+- The cluster should be reachable via at http://localhost:9200 ("You Know, for Search").
 
-_To see log output, run `docker-compose logs usaspending-es`_
+- Optionally, to see log output, use `docker-compose logs usaspending-es` (these logs are stored by docker even if you don't use this).
 
-The cluster should be reachable via `curl` or Kibana at http://localhost:9200
+- To populate this cluster's index(es) with data from your database, you would need to run the Django Managment command in `es_rapidloader.py`, which can be run from Docker directly, after setting `ES_HOSTNAME` in `.env`:
```
Comment (tony-sappe) [inline README.md]: `es_rapidloader.py` is worthless without data in Postgres. I recommend we leave this note out to reduce confusion.

Comment (willkjackson) [reply README.md]: I'm adding a data subset restoration step.

Comment (willkjackson) [reply README.md]: Scratch that, I added a ticket to add "restoring a subset to psql" and "restoring ES index (from the subset)" to this document. Removing this.

File: .env [hunk 4]
```diff
@@ -16,7 +16,7 @@ MATVIEW_SQL_DIR=matview_sql
 
 # Where to connect to elasticsearch.
 # Should include scheme (http:// or https://, host, and port (if different than 80 or 443 for HTTP/HTTPS)
-ES_HOSTNAME=http://usaspending-es:9200
+ES_HOSTNAME=http://localhost:9200
```
Comment (tony-sappe) [inline .env]: @willkjackson the ES_HOSTNAME env var is for the USAspending API container

**Pull request merged**